### ISSUE TITLE
Spike: Forall expression failures under valgrind - 1

### ISSUE THREAD

**Issue by vasslitvinov** [chapel-lang/chapel#10477]
This task is to fix the valgrind failure due to #9994 in the test:
```
test/arrays/shapes/skyline.chpl
```

This is a non-skyline version of the above, which also needs to work:

```chpl
var D1 = {1..3};
var D2 = {1..2};
var A = [idx1 in D1] [idx2 in D2] idx1*100 + idx2;

compilerWarning(A.type:string);
compilerWarning(A.eltType:string);
writeln(A.domain);
writeln(A[1].domain);
writeln(A);
```

**Comment by bradcray** [comment]
What's the story here?

**Comment by vasslitvinov** [comment]
The problem is described and a solution is implemented in #11043.

There is a bit of confusion in that this test is now exhibiting a separate issue, #10956.
